crates/api/src/relay_data/api.rs

```diff
@@ -28,23 +29,22 @@ pub(crate) const PATH_VALIDATOR_REGISTRATION: &str = "/validator_registration";
 pub(crate) type BidsCache = Cache<String, Vec<ReceivedBlocksResponse>>;
 pub(crate) type DeliveredPayloadsCache = Cache<String, Vec<DeliveredPayloadsResponse>>;
 
-#[derive(Clone)]
 pub struct DataApi<A:Auctioneer, DB: DatabaseService> {
     validator_preferences: Arc<ValidatorPreferences>,
     auctioneer: Arc<A>,
     db: Arc<DB>,
 
-    constraints_rx: Arc<RwLock<mpsc::Receiver<ConstraintsMessage>>>,
+    constraints_rx: broadcast::Receiver<ConstraintsMessage>,
 }
 
 impl<A: Auctioneer + 'static, DB: DatabaseService + 'static> DataApi<A, DB> {
     pub fn new(validator_preferences: Arc<ValidatorPreferences>, auctioneer:Arc<A>, db: Arc<DB>) -> (Self, ConstraintsHandle) {
-        let (constraints_tx, constraints_rx) = mpsc::channel(100); // Should we have buffer?
+        let (constraints_tx, constraints_rx ) = broadcast::channel(100); // Should we have buffer?
```

Comment:
to address the comment, I would leave the channel to something like 128 constraint messages - it's not likely to ever get filled anyway